src/baryvel.jl

```diff
@@ -0,0 +1,229 @@
+                 1.362948  -1.1790629318198e3 ;
+                 5.5657014 -1.0977134971135e3 ;
+                 5.0708205 -1.5774000881978e2 ;
+                 3.9318944  5.296346478e1     ;
+                 4.8989497  3.9809289073258e1 ;
+                 1.3097446  7.7540959633708e1 ;
+                 3.5147141  7.9618578146517e1 ;
+                 3.5413158 -5.4868336758022e2 ]
+
+const ccamps = [-2.279594e-5  1.407414e-5  8.273188e-6  1.340565e-5 -2.490817e-7 ;
+                -3.494537e-5  2.860401e-7  1.289448e-7  1.627237e-5 -1.823138e-7 ;
+                 6.593466e-7  1.322572e-5  9.258695e-6 -4.674248e-7 -3.646275e-7 ;
+                 1.140767e-5 -2.049792e-5 -4.747930e-6 -2.638763e-6 -1.245408e-7 ;
+                 9.516893e-6 -2.748894e-6 -1.319381e-6 -4.549908e-6 -1.864821e-7 ;
+                 7.310990e-6 -1.924710e-6 -8.772849e-7 -3.334143e-6 -1.745256e-7 ;
+                -2.603449e-6  7.359472e-6  3.168357e-6  1.119056e-6 -1.655307e-7 ;
+                -3.228859e-6  1.308997e-7  1.013137e-7  2.403899e-6 -3.736225e-7 ;
+                 3.442177e-7  2.671323e-6  1.832858e-6 -2.394688e-7 -3.478444e-7 ;
+                 8.702406e-6 -8.421214e-6 -1.372341e-6 -1.455234e-6 -4.998479e-8 ;
+                -1.488378e-6 -1.251789e-5  5.226868e-7 -2.049301e-7  0           ;
+                -8.043059e-6 -2.991300e-6  1.473654e-7 -3.154542e-7  0           ;
+                 3.699128e-6 -3.316126e-6  2.901257e-7  3.407826e-7  0           ;
+                 2.550120e-6 -1.241123e-6  9.901116e-8  2.210482e-7  0           ;
+                -6.351059e-7  2.341650e-6  1.061492e-6  2.878231e-7  0           ]
+
+const ccsec = [ 1.289600e-6 5.550147e-1  2.076942    ;
+                3.102810e-5 4.035027     3.525565e-1 ;
+                9.124190e-6 9.990265e-1  2.622706    ;
+                9.793240e-7 5.508259     1.559103e01 ]
+
+const dcargm = [5.167983  8.3286911095275e3 ;
+                5.491315 -7.2140632838100e3 ;
+                5.959853  1.5542754389685e4 ]
+
+const ccampm = [ 1.097594e-1 2.896773e-7 5.450474e-2  1.438491E-7 ;
+                -2.223581e-2 5.083103e-8 1.002548e-2 -2.291823E-8 ;
+                 1.148966e-2 5.658888e-8 8.249439e-3  4.063015E-8 ]
+
+const ccpamv = [8.326827e-11, 1.843484e-11, 1.988712e-12, 1.881276e-12]
+
+function _baryvel(dje::T, deq::T) where {T<:AbstractFloat}
+    # Time arguments.
+    dt = (dje - 2415020) / JULIANCENTURY
+    tvec = [1; dt; dt*dt]
+
+    temp = rem.(dcfel * tvec, 2 * T(pi))
+    dml, g = temp[1:2]
+    forbel = temp[2:8]
+    deps = rem.(dot(tvec, [0.4093198; -2.271110e-4; -2.860401e-8]), 2 * T(pi))
+    sorbel = rem.(ccsel * tvec, 2 * T(pi))
+    e = sorbel[1]
+
+    # Secular perturbations in longitude.
+    sn = sin.(rem.(ccsec[:,2:3] * tvec[1:2], 2 * T(pi)))
+
+    # Periodic perturbations of the emb (earth-moon barycenter).
+    pertl = dot(ccsec[:,1], sn) + dt * sn[3] * -7.757020e-8
+    a = rem.(dcargs[:,1] + dt * dcargs[:,2], 2 * T(pi))
```

Comment:
Dot-call syntax is very handy and powerful to avoid explicit loops, but here an explicit loops is used anyway.  I'd move `a` inside the loop (in this case it would be a simple scalar variable),, just like in the IDL version, and you can compute `cosa` and `sina` at once with the [`sincos`](https://docs.julialang.org/en/latest/stdlib/math/#Base.Math.sincos) function (I introduced it for Julia 0.6 with https://github.com/JuliaAstro/AstroLib.jl/commit/68e1a2a9d1367f0cc7217d373c516e2294364ece):
```julia
sina, cosa = sincos(a)
```